### PR TITLE
Fix sorting on 'closing soonest' when getting open markets from Omen

### PR DESCRIPTION
**Pull request by evangriffiths**
## Summary by CodeRabbit

- **New Features**
	- Introduced a new attribute to track market opening times more accurately.
	- Enhanced sorting functionality for markets to prioritize by closing time.

- **Bug Fixes**
	- Adjusted market close time handling to improve accuracy in market status.

- **Tests**
	- Added tests to ensure correct handling of market close times and betting strategies.

### CODE REVIEW

File: tests/markets/omen/test_omen.py [hunk 1]
```diff
@@ -152,3 +152,10 @@ def create_market_fund_market_remove_funding() -> None:
         - Assert amount in xDAI is reflected in user's balance
     """
     assert True
+
+
+def test_omen_market_close_time() -> None:
+    market = OmenAgentMarket.from_data_model(pick_binary_market())
+    assert (
+        market.close_time > market.created_time
+    ), "Market close time should be after open time."
```
Comment (gabrielfior) [inline tests/markets/omen/test_omen.py:161]: I would also test the sorting part of things that you added - make sure that all elements in the list are properly sorted is trivial, but improves coverage of the codebase.

File: prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py [hunk 2]
```diff
@@ -201,8 +201,7 @@ def _build_sort_direction(self, sort_by: SortBy) -> str:
             case SortBy.NEWEST:
                 sort_direction = "desc"
             case SortBy.CLOSING_SOONEST:
-                # `desc` feel unintuitive, but really if we use `asc`, we are getting markets closing in 2030, 2026, etc.
-                sort_direction = "desc"
+                sort_direction = "asc"
```
Comment (kongzii) [inline prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py:204]: wdyt about `self._build_sort_direction` returning both the sorting field and sorting direction? something like `return self.trades_subgraph.FixedProductMarketMaker.openingTimestamp, "asc"`, it should be easier and less error-prone then to extend in the future

File: prediction_market_agent_tooling/markets/omen/data_models.py [hunk 3]
```diff
@@ -112,6 +112,9 @@ class OmenMarket(BaseModel):
     creationTimestamp: t.Optional[int] = None
     condition: Condition
     question: Question
+    openingTimestamp: t.Optional[
```
Comment (evangriffiths) [inline prediction_market_agent_tooling/markets/omen/data_models.py:115]: note: for some reason `question.openingTimestamp` returns None for all fpmms, whereas this returns a value as expected (even though they should be the same!), so using this field

File: tests/markets/omen/test_omen.py (hunk 1 above)
Comment (evangriffiths) [reply tests/markets/omen/test_omen.py]: Good point, added

File: prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py [hunk 4]
```diff
@@ -176,6 +172,7 @@ def _build_where_statements(
         if resolved is not None:
             if resolved:
                 where_stms["resolutionTimestamp_not"] = None
+                where_stms["currentAnswer_not"] = self.INVALID_ANSWER
```
Comment (evangriffiths) [inline prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py:175]: Existing bugfix - resolved markets should not have invalid answers

Comment (kongzii) [reply prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py:175]: Actually, resolved market can have invalid answer:

Screenshot by Dropbox Capture

Market can be resolved as INVALID if we can't determine the correct answer.

Comment (evangriffiths) [reply prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py:175]: Do we want to have these markets returned in our API though? If so, the test prediction-market-agent-tooling/tests/markets/test_omen_subgraph_handler.py::test_filter_resolved_markets needs updating instead.

File: prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py (hunk 2 above)
Comment (evangriffiths) [reply prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py:208]: Agreed, fixed

File: prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py (hunk 4 above)
Comment (kongzii) [reply prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py:175]: I thought it's correct to return them as they are resolved markets, but @gabrielfior had a similar comment when you were on vacation.

So, if you both think it's better just to filter them out, let's keep it.

**Review by kongzii — APPROVED** [review]

File: prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py (hunk 4 above)
Comment (evangriffiths) [reply prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py:175]: Okay. I guess from what I can see we don't have a use-case for markets resolved with an invalid answer. If we do, then we can change.

**Pull request merged**